File: keiner_lindarte_aro/instructor_henry/mis_programas/ruby/orientado_a_objetos/5_clase_de_09_agosto/sobreescribir_metodo.rb
class Humano
  attr_accessor :nombre, :sexo, :nacionalidad

  def initialize(sexo, nacionalidad, nombre)
    self.sexo = sexo
    self.nacionalidad = nacionalidad
    self.nombre = nombre
  end


  # modificamos el metodo length
  def length
    puts "Se ha creado un nuevo un humano de genero #{self.sexo}, con el nombre #{self.nombre} y de nacionalidad #{self.nacionalidad}"
  end

end

nuevo1 = Humano.new("hombre","Keiner lindarte","Colombiano")
puts nuevo1.length

File: keiner_lindarte_aro/instructor_henry/mis_programas/dart/orientado_a_objetos/5_clase_de_09_agosto/sobreescribir_metodo.dart
class Humano {
  String? nombre;
  String? sexo;
  String? nacionalidad;

  Humano({this.sexo, this.nacionalidad, this.nombre});

  @override
  String toString() {
    return "Se ha creado un nuevo un humano de genero ${sexo}, con el nombre ${nombre} y de nacionalidad ${nacionalidad}";
  }
}

void main(){

  Humano nuevo1 = new Humano(sexo: "hombre", nombre: "Keiner lindarte", nacionalidad: "colombiano");
  print(nuevo1.toString());

}